### PR TITLE
BREAKING CHANGE(expo-calendar): forbid passing an id to createEventAsync and updateEventAsync

### PR DESCRIPTION
**Pull request by pierrezimmermannbam**
# Why

I noticed that the `eventData` param of the `createEventAsync` method from `expo-calendar` can have an `id` property but it's not used in the code. What I found out is that we the id will be generated by the OS calendar system so we can't choose it. I think the typing should reflect the fact that id is not used so it should not be allowed to pass an `eventData` object with an `id` property. This is also true for the `updateEventAsync` method.

This is a breaking change only when these methods are used with object litterals

```tsx
// This will work
const eventData = {
  title: 'title',
  id: '1',
  }
updateEventAsync(eventId, eventData);

// This will trigger an error
updateEventAsync(eventId, {
  title: 'title',
  id: '1',
  });

```

So it doesn't fully prevent passing ids but the type and the docs will indicate that these methods don't do anything with an id if provided

# How

I updated the typing of `eventData` param of `createEventAsync` and `updateEventAsync` methods to `Omit<Partial<Event>, 'id'>`  

# Test Plan

I checked that there was no typescript error in the places where these methods are used. 

# Checklist

Regarding documentation update I'm not very sure. From what I have understood it should update itself automatically based on the API so there should not be any change needed. When running the docs website locally though I didn't see my changes but I guess it doesn't use my local version?

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [X] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [X] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-calendar/src/Calendar.ts [hunk 1]
```diff
@@ -757,19 +757,17 @@ export async function createEventAsync(
     throw new Error('createEventAsync must be called with an id (string) of the target calendar');
   }
 
-  const { id, ...details } = eventData;
```
Comment (lukmccall) [inline packages/expo-calendar/src/Calendar.ts:760]: Let's leave that logic for now. Some users don't use typescript, so they won't know about this breaking change. However, we should add a runtime warning if the `id` is defined. What do you think about it?

**Review by lukmccall — CHANGES_REQUESTED** [review]

File: packages/expo-calendar/src/Calendar.ts (hunk 1 above)
Comment (pierrezimmermannbam) [reply packages/expo-calendar/src/Calendar.ts:760]: Oh right good catch I hadn't thought about that, I'll make the change

Comment (pierrezimmermannbam) [reply packages/expo-calendar/src/Calendar.ts:760]: I put back the logic. I had to use a ts-expect-error though because TS complained `id` wasn't defined

**Review by lukmccall — APPROVED** [review]
Thanks for applying my suggestion ✅
I'll merge your PR as soon as the CI jobs pass.

**Comment by pierrezimmermannbam** [comment]
Thanks! I'm currently on vacations but I'll fix that when I return

**Pull request merged**